File: otto-android/app/src/main/java/to/ottomot/driftd/ui/squad/SquadInviteAcceptDialog.kt
package to.ottomot.driftd.ui.squad

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Groups
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import to.ottomot.driftd.R
import to.ottomot.driftd.core.network.dto.InviteLinkResolveDto
import to.ottomot.driftd.ui.dialog.OttoEducationDialog

@Composable
fun SquadInviteAcceptDialog(
    resolve: InviteLinkResolveDto,
    isAccepting: Boolean,
    onAccept: () -> Unit,
    onDecline: () -> Unit,
) {
    val squadName =
        resolve.circle?.name?.trim()?.takeIf { it.isNotEmpty() }
            ?: stringResource(R.string.squad_invite_default_squad_name)
    val inviterName =
        resolve.invitedBy?.displayName?.trim()?.takeIf { it.isNotEmpty() }
            ?: stringResource(R.string.squad_invite_default_inviter_name)

    OttoEducationDialog(
        visible = true,
        busy = isAccepting,
        onDismissRequest = onDecline,
        onCloseClick = onDecline,
        hero = {
            Box(
                modifier =
                    Modifier
                        .size(72.dp)
                        .background(Color(0x387B3DFF), CircleShape),
                contentAlignment = Alignment.Center,
            ) {
                Icon(
                    imageVector = Icons.Filled.Groups,
                    contentDescription = null,
                    tint = Color(0xFF7B3DFF),
                    modifier = Modifier.size(28.dp),
                )
            }
        },
        title = stringResource(R.string.squad_invite_prompt_title, squadName),
        body = stringResource(R.string.squad_invite_prompt_body, inviterName, squadName),
        bulletSectionTitle = null,
        bullets = emptyList(),
        footer = stringResource(R.string.squad_invite_prompt_footer),
        primaryLabel = stringResource(R.string.squad_invite_prompt_accept),
        onPrimaryClick = onAccept,
        secondaryLabel = stringResource(R.string.squad_invite_prompt_decline),
        onSecondaryClick = onDecline,
        allowsUnconfirmedDismiss = true,
    )
}

File: otto-mobile/otto-mobile/SquadInviteAcceptDialog.swift
import SwiftUI

/// Confirmation sheet after opening a squad invite link (`driftd.com/invite/{code}`).
struct SquadInviteAcceptDialog: View {
    let resolve: InviteLinkResolveDTO
    var isAccepting: Bool
    var onAccept: () -> Void
    var onDecline: () -> Void

    private var squadName: String {
        resolve.circle?.name ?? String(localized: "squad_invite_default_squad_name")
    }

    private var inviterName: String {
        resolve.invitedBy?.displayName ?? String(localized: "squad_invite_default_inviter_name")
    }

    var body: some View {
        OttoEducationDialog(
            allowsUnconfirmedDismiss: true,
            onDismissUnconfirmed: onDecline,
            hero: {
                OttoEducationSquadHero()
            },
            title: String(format: String(localized: "squad_invite_prompt_title"), squadName),
            bodyText: String(format: String(localized: "squad_invite_prompt_body"), inviterName, squadName),
            bulletSectionTitle: nil,
            bullets: [],
            footer: String(localized: "squad_invite_prompt_footer"),
            primaryTitle: String(localized: "squad_invite_prompt_accept"),
            onPrimary: onAccept,
            secondaryTitle: String(localized: "squad_invite_prompt_decline")
        )
        .overlay {
            if isAccepting {
                ProgressView()
                    .tint(.white)
            }
        }
        .allowsHitTesting(!isAccepting)
    }
}

private struct OttoEducationSquadHero: View {
    var body: some View {
        ZStack {
            Circle()
                .fill(Color(red: 0.482, green: 0.239, blue: 1.0).opacity(0.22))
                .frame(width: 72, height: 72)
            Image(systemName: "person.3.fill")
                .font(.system(size: 28, weight: .semibold))
                .foregroundStyle(Color(red: 0.482, green: 0.239, blue: 1.0))
        }
    }
}
